example/spec/parallel/ingestGranule/IngestGranuleSuccessSpec.js

```diff
@@ -437,34 +392,59 @@ describe('The S3 Ingest Granules workflow', () => {
         expect(granule.cmrLink).not.toBeUndefined();
       });
 
-      it('allows reingest and executes with success status', async () => {
-        const oldUpdatedAt = granule.updatedAt;
-        const oldExecution = granule.execution;
+      describe('reingest granule api', () => {
+        let oldExecution;
+        let oldUpdatedAt;
+        let reingestResponse;
+        let startTime;
+
+        beforeAll(async () => {
+          startTime = new Date();
+          oldUpdatedAt = granule.updatedAt;
+          oldExecution = granule.execution;
+          const reingestGranuleResponse = await apiTestUtils.reingestGranule({
+            prefix: config.stackName,
+            granuleId: inputPayload.granules[0].granuleId
+          });
+          reingestResponse = JSON.parse(reingestGranuleResponse.body);
+          console.log(`reingest granule response: ${JSON.stringify(reingestResponse)}`);
+        });
 
-        // Reingest Granule and compare the updatedAt times
-        const reingestGranuleResponse = await apiTestUtils.reingestGranule({
-          prefix: config.stackName,
-          granuleId: inputPayload.granules[0].granuleId
+        it('executes with success status', () => {
+          expect(reingestResponse.status).toEqual('SUCCESS');
         });
-        const reingestResponse = JSON.parse(reingestGranuleResponse.body);
-        expect(reingestResponse.status).toEqual('SUCCESS');
 
-        const secondGranuleResponse = await apiTestUtils.getGranule({
-          prefix: config.stackName,
-          granuleId: inputPayload.granules[0].granuleId
+        it('returns a warning that data may be overwritten when duplicateHandling is "error"', () => {
+          expect(reingestResponse.warning && reingestResponse.warning.includes('overwritten')).toBeTruthy();
         });
-        const secondGranule = JSON.parse(secondGranuleResponse.body);
-        expect(secondGranule.updatedAt).toBeGreaterThan(oldUpdatedAt);
 
-        // Await reingest completion
-        await waitUntilGranuleStatusIs(config.stackName, inputPayload.granules[0].granuleId, 'completed');
-        const updatedGranuleResponse = await apiTestUtils.getGranule({
-          prefix: config.stackName,
-          granuleId: inputPayload.granules[0].granuleId
+        it('overwrites granule files', async () => {
+          // Await reingest completion
+          await waitUntilGranuleStatusIs(config.stackName, inputPayload.granules[0].granuleId, 'completed');
```

Comment:
It might be a good idea to wait for the workflow to complete before doing checks. I say this because I fixed a race condition on a reingest last week. See line 478.

```
const publishGranuleExecution = await waitForTestExecutionStart(
          'PublishGranule',
          config.stackName,
          config.bucket,
          isExecutionForGranuleId,
          { granuleId: inputPayload.granules[0].granuleId }
        );
```

Reply:
The workflow name for reingest is not available here, it is from the granule's previous execution.  If it's necessary, I can get that workflow name from that execution.

Reply:
Wouldn't we know it's the workflow we've just run though? `IngestAndPublishGranule` i guess?

Reply:
Actually, in this case, it would be `IngestAndPublishGranule`.